### PR TITLE
Added dashboard variables for Traefik.

### PR DESCRIPTION
**Pull request by miransar**
# Summary

- Added dashboard variables for Traefik
- Added Prometheus Agent install plan

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include an InstallPlan and Documentation reference?
- [ ] Are all documentation links publically accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [ ] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

### CODE REVIEW

File: dashboards/traefik/traefik.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nmax(traefik_config_last_reload_success)*1000 AS 'Last Reload Success',\nmax(traefik_config_last_reload_failure)*1000 AS 'Last Reload Failure',\nsum(traefik_config_reloads_total) AS 'Total Reloads',\nsum(traefik_config_reloads_failure_total) AS 'Total Failed Reloads'\nFACET prometheus_server, instance\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nmax(traefik_config_last_reload_success)*1000 AS 'Last Reload Success',\nmax(traefik_config_last_reload_failure)*1000 AS 'Last Reload Failure',\nsum(traefik_config_reloads_total) AS 'Total Reloads',\nsum(traefik_config_reloads_failure_total) AS 'Total Failed Reloads'\nFACET prometheus_server, instance\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:54]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance'
```

File: dashboards/traefik/traefik.json [hunk 2]
```diff
@@ -78,7 +78,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nsum(traefik_entrypoint_requests_total) AS 'Total Requests'\nFACET instance, entrypoint\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nsum(traefik_entrypoint_requests_total) AS 'Total Requests'\nFACET instance, entrypoint\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:81]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', entrypoint
```

File: dashboards/traefik/traefik.json [hunk 3]
```diff
@@ -108,7 +108,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nlatest(traefik_entrypoint_open_connections) AS 'Open Connections'\nFACET instance, entrypoint\nWHERE instrumentation.name = 'remote-write'\nTIMESERIES\nLIMIT MAX"
+                "query": "FROM Metric SELECT\nlatest(traefik_entrypoint_open_connections) AS 'Open Connections'\nFACET instance, entrypoint\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance) \nTIMESERIES\nLIMIT MAX "
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:111]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', entrypoint
```

File: dashboards/traefik/traefik.json [hunk 4]
```diff
@@ -138,7 +138,7 @@
...
+                "query": "FROM Metric SELECT\nlatest(traefik_entrypoint_open_connections) AS 'Open Connections',\nsum(traefik_entrypoint_requests_total) AS 'Total Requests',\nsum(traefik_entrypoint_requests_tls_total) AS 'Total TLS Requests',\nbucketPercentile(traefik_entrypoint_request_duration_seconds_bucket, 95) AS 'Request Duration (ms)',\naverage(traefik_entrypoint_request_duration_seconds_sum) AS 'Average Request Duration (ms)'\nFACET prometheus_server, instance, entrypoint, method, protocol\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:141]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', entrypoint, method, protocol
```

File: dashboards/traefik/traefik.json [hunk 5]
```diff
@@ -165,7 +165,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nsum(traefik_service_requests_total) AS 'Total Requests'\nFACET instance, service\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nsum(traefik_service_requests_total) AS 'Total Requests'\nFACET instance, service\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:168]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', service
```

File: dashboards/traefik/traefik.json [hunk 6]
```diff
@@ -195,7 +195,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT count(traefik_service_requests_total) AS 'Total Requests' FACET instance, service, code WHERE instrumentation.name = 'remote-write' LIMIT MAX"
+                "query": "FROM Metric SELECT count(traefik_service_requests_total) AS 'Total Requests' FACET instance, service, code WHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance) LIMIT MAX"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:198]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', service, code
```

File: dashboards/traefik/traefik.json [hunk 7]
```diff
@@ -222,7 +222,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\naverage(traefik_service_request_duration_seconds_sum) AS 'Average Request Duration (ms)'\nFACET instance, service\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\naverage(traefik_service_request_duration_seconds_sum) AS 'Average Request Duration (ms)'\nFACET instance, service\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:225]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', service
```

File: dashboards/traefik/traefik.json [hunk 8]
```diff
@@ -249,7 +249,7 @@
...
-                "query": "FROM Metric SELECT\nlatest(traefik_service_open_connections) AS 'Open Connections',\nsum(traefik_service_requests_total) AS 'Total Requests',\nsum(traefik_service_requests_tls_total) AS 'Total TLS Requests',\nbucketPercentile(traefik_service_request_duration_seconds_bucket, 95) AS 'Request Duration (ms)',\naverage(traefik_service_request_duration_seconds_sum) AS 'Average Request Duration (ms)'\nFACET prometheus_server, service, method, protocol\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nlatest(traefik_service_open_connections) AS 'Open Connections',\nsum(traefik_service_requests_total) AS 'Total Requests',\nsum(traefik_service_requests_tls_total) AS 'Total TLS Requests',\nbucketPercentile(traefik_service_request_duration_seconds_bucket, 95) AS 'Request Duration (ms)',\naverage(traefik_service_request_duration_seconds_sum) AS 'Average Request Duration (ms)'\nFACET prometheus_server, service, method, protocol\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:252]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', service, method, protocol
```

File: dashboards/traefik/traefik.json [hunk 9]
```diff
@@ -282,7 +282,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nlatest(traefik_tls_certs_not_after)*1000 AS 'Expiration Date'\nFACET prometheus_server, instance, cn AS 'Common Name', sans AS 'Subject Alternate Name', serial\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nlatest(traefik_tls_certs_not_after)*1000 AS 'Expiration Date'\nFACET prometheus_server, instance, cn AS 'Common Name', sans AS 'Subject Alternate Name', serial\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} ) facet concat(prometheus_server,' - ',instance)"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:285]: 2 `FACET` conditions in this query; recommend consolidating into:

```
FACET concat(prometheus_server,' - ',instance) AS 'Instance', cn AS 'Common Name', sans AS 'Sunject Alternate Name', serial
```

**Review by thezackm — CHANGES_REQUESTED** [review]
@miransar - this dashboard has widgets with 2 `FACET` conditions which will error out on import; comments have been added with recommendations for `prometheus-growth` project

File: dashboards/traefik/traefik.json (hunk 2 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json (hunk 3 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json (hunk 4 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json (hunk 5 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json (hunk 7 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json (hunk 8 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Thanks, changes made.

File: dashboards/traefik/traefik.json [hunk 10]
```diff
@@ -282,7 +282,7 @@
             "nrqlQueries": [
               {
                 "accountId": 0,
-                "query": "FROM Metric SELECT\nlatest(traefik_tls_certs_not_after)*1000 AS 'Expiration Date'\nFACET prometheus_server, instance, cn AS 'Common Name', sans AS 'Subject Alternate Name', serial\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write'"
+                "query": "FROM Metric SELECT\nlatest(traefik_tls_certs_not_after)*1000 AS 'Expiration Date'\nFACET concat(prometheus_server,' - ',instance) AS 'Instance', cn AS 'Common Name', sans AS 'Sunject Alternate Name', serial\nLIMIT MAX\nWHERE instrumentation.name = 'remote-write' AND prometheus_server IN ( {{prometheus_server}}) AND instance IN ( {{instance}} )"
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:285]: typo here:

`Sunject Alternate Name` should be `Subject Alternate Name`

**Review by thezackm — CHANGES_REQUESTED** [review]
@miransar - one small typo - the rest looks solid

**Review by thezackm — APPROVED** [review]
approved for `prometheus-growth` project

File: dashboards/traefik/traefik.json (hunk 10 above)
Comment (miransar) [reply dashboards/traefik/traefik.json]: Fixed. Thanks.

File: dashboards/traefik/traefik.json [hunk 11]
```diff
@@ -293,5 +293,48 @@
...
+        {
+          "value": {
+            "string": "*"
+          }
+        }
+      ],
+      "nrqlQuery": {
+        "accountIds": [
+          0
+        ],
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:310]: set to `"accountIds": []`

File: dashboards/traefik/traefik.json [hunk 12]
```diff
@@ -293,5 +293,48 @@
...
```
Comment (thezackm) [inline dashboards/traefik/traefik.json:331]: set to `"accountIds": []`

File: quickstarts/traefik/config.yml [hunk 13]
```diff
@@ -32,11 +33,16 @@ keywords:
...
+      Use Prometheus remote_write or Prometheus Agent
+    url: https://docs.newrelic.com/docs/infrastructure/prometheus-integrations/get-started/send-prometheus-metric-data-new-relic
+  - name: Traefik Documentations
+    description: |
+      Traefik Integration Documentation
+    url: https://docs.newrelic.com/docs/infrastructure/prometheus-integrations/integrations-list/traefik-integration
 
 installPlans:
   - setup-prometheus
+  - setup-prometheus-agent-mode
```
Comment (thezackm) [inline quickstarts/traefik/config.yml:45]: pending in https://github.com/newrelic/newrelic-quickstarts/pull/1503

**Review by thezackm — CHANGES_REQUESTED** [review]
@miransar - changes requested; ty

File: dashboards/traefik/traefik.json (hunk 11 above)
Comment (miransar) [reply dashboards/traefik/traefik.json:310]: Changed made. Thanks.

File: dashboards/traefik/traefik.json (hunk 12 above)
Comment (miransar) [reply dashboards/traefik/traefik.json:331]: Changed made. Thanks.

**Review by thezackm — APPROVED** [review]

**Review by josephgregoryii — APPROVED** [review]

**Pull request merged**